packages/aws-cdk/lib/cli.ts

```diff
@@ -514,7 +514,7 @@ export async function exec(args: string[], synthesizer?: Synthesizer): Promise<n
           stream: args.ci ? process.stdout : undefined,
           compareAgainstProcessedTemplate: args.processed,
           quiet: args.quiet,
-          changeSet: args['change-set'],
+          changeSet: configuration.settings.get(['diff', 'changeSet']) ?? args['change-set'],
```

Comment:
```suggestion
          changeSet: args['change-set'] ?? configuration.settings.get(['diff', 'changeSet'],
```

Reply:
I think you are right about this order of operations. Only question is if there is a difference between no flag being provided and the default. 

If the flag is always given a value regardless of being set, we may have to put the context first (but I will ask some others on the team about this).